hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -645,15 +604,50 @@ public void run() {
             }
 
             try {
-                authenticate(target, connection, asOwner, future);
+                authenticateAsync();
```

Comment:
I still prefer the `connection` passed as a parameter over making it field of the class. That way it is easier to follow the logic. When it is a class field one needs to be aware of all the methods because it can be assigned/changed in any of the methods.